README.md

```diff
@@ -121,72 +268,108 @@ User = db.define ('User', {
+
+`automigrate` destroys existing model instances if model already defined in database. Please make sure you do want to clean up data before running `automigrate`. Then it does same thing as `autoupdate`
+
+## isActual
+
+User can call this function to check if model exists in database.
+
+## example code
+
+*/server/script.js*
+```javascript
+module.export = function migrateData(app) {
+  var ds = app.models.cloudantDS;
+  
+  // static model created with model.json file
+  var StaticModel = app.models.StaticModel;
+  // dynamic model created in boot script
+  var DynamicModel = ds.define('DynamicModel', {
+      name: {type: String},
+      description: {type: String},
+  });
+
+  // Write the three examples in parallel just to avoid dup code,
+  // please try ONLY ONE of them at one time.
+  ds.once('connected', function() {
+    // try autoupdate example - multiple models
+    ds.autoupdate(['StaticModel', 'DynamicModel'], function(err) {});
+    // OR
+    // try automigrate example - single model
+    ds.automigrate('StaticModel', function(err) {});
+    // OR
+    // try isActual example - if any model exist, run autoupdate, otherwise automigrate
+    ds.isActual(['StaticModel', 'DynamicModel'], function(err, exist) {
+      if (exist) {
+        ds.autoupdate(['StaticModel', 'DynamicModel'], function(err){})
+      } else {
+        ds.automigate(['StaticModel', 'DynamicModel'], function(err){});
+      }
+    });
+  });
 }
 ```
 
-**Please note how property `prop2` was completely dropped upon update.**
+# Discovery
 
-**Solution:** Do not pass partial values for the data object to be updated. If there are properties that are not being updated, please include the old value to keep data persistent.
+Not implemented yet, track it in story https://github.com/strongloop/loopback-connector-cloudant/issues/118 
 
-## Feature backlog
+# Query
 
-* Index-only model properties marked with index=true
-* Configurable "view based" or JSON indexes. [More Info>>](https://cloudant.com/blog/mango-json-vs-text-indexes)
+-  Uses Cloudant Query (Lucene) to support ad-hoc searching.
+-  [Loopback query](http://loopback.io/doc/en/lb3/Querying-data.html) support for: fields, limit, order, skip and where filters.
+- Please check [Advanced Queries](https://github.com/strongloop/loopback-connector-cloudant/blob/master/doc/advanced-queries.md) for details about regex filter, nested filter and order.
 
-## Setup Cloudant instance
+# Setup Cloudant instance
 
-There is no free version of local Cloudant to download, so to develop or test with Cloudant connector, you must set up either a Cloudant DBaas instance or a Bluemix Cloudant instance.
+For development use, a docker container of cloudant local is easy to setup and there is no request limit per second.
+Bluemix cloudant will be more stable for product.
```

Comment:
@0candy I thought they mean the same thing when use "product" as an noun..? I will change it to "production" :)